### DISCUSSION TITLE
Multi-Select Product Attribute Backend Change

### DISCUSSION THREAD

**Discussion by BrianPittVP** [OpenMage/magento-lts#2655] [Questions & Answers] [answered]
Somewhere in the `1.9.3.x` release, Magento changed how it stores Multi-Select product attributes.  They used to be stored in the `catalog_product_entity_varchar` table, but they moved it to `catalog_product_entity_text`, I assume to allow for more space.

We recently underwent an upgrade from **Magento 1.9.2.4** to **OpenMage v20.x**, and one of the things we noticed is that some attributes' values have moved to the `catalog_product_entity_text` table.  Everything is working fine within OpenMage, we caught this because we had external scripts hard-coded to grab values that suddenly disappeared.

I'm trying to understand when and how this happens.  I don't see anything in the code that migrates data from one to the other, or anything that looks in one table if it doesn't find a value in the other.  Admittedly, I haven't gone through every single line of code, but I've searched for everything I can think of, and nothing jumps out.

Does anyone understand this enough to explain it, or at least point me in the right direction?

**Comment by kiatng** [comment, marked as answer]
Using VS Code, I found line 42 in seconds:

![image](https://user-images.githubusercontent.com/1106470/195737937-85698477-9673-4086-aa7f-093716b65f7a.png)

The commit is here:
https://github.com/OpenMage/magento-lts/commit/d48bebc211cc216aaf78bdf25d7f0b0143d6333b

In your browser, search for _multiselect_, I think you will find your answers.

**Reply by BrianPittVP** [reply]
Thanks.  I didn't know what to look for, I appreciate the 42 seconds of your time.
